src/providers/mod.rs

```diff
@@ -69,7 +69,7 @@ impl<'s> Providers<'s> {
         }
 
         Providers {
-            default_provider: &settings.provider,
+            default_provider: settings.provider.to_owned(),
```

Comment:
Not that it makes much/any difference, but `to_owned` feels mildly suprising as a reader given that `settings.provider` is already a `String` rather than a `&str`. Would `.cloned()` be clearer?